surefire-provider/src/test/java/org/arquillian/smart/testing/surefire/provider/info/CustomProviderInfoTest.java

```diff
@@ -0,0 +1,57 @@
+package org.arquillian.smart.testing.surefire.provider.info;
+
+import java.io.IOException;
+import org.arquillian.smart.testing.hub.storage.local.TemporaryInternalFiles;
+import org.arquillian.smart.testing.surefire.provider.LoaderVersionExtractor;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static org.arquillian.smart.testing.known.surefire.providers.KnownProvider.JUNIT_5;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CustomProviderInfoTest {
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private final String junit5Coordinates = String.join(":", JUNIT_5.getGroupId(), JUNIT_5.getArtifactId(), "1.0.1");
+
+    @Test
+    public void should_read_junit_5_platform_version_when_failsafe_used() throws IOException {
+        // given
+        createCustomProviderInfoFileWithJunit5(LoaderVersionExtractor.ARTIFACT_ID_MAVEN_FAILSAFE_PLUGIN);
+        CustomProviderInfo customProviderInfo = new CustomProviderInfo(temporaryFolder.getRoot());
+
+        // when
+        customProviderInfo.retrieveCustomProviderInformation(temporaryFolder.getRoot(), "2.20");
+
+        // then
+        verifyCustomProviderInfoHasJunit5(customProviderInfo);
+    }
+
+    @Test
+    public void should_read_junit_5_platform_version_when_surefire_used() throws IOException {
+        // given
+        createCustomProviderInfoFileWithJunit5(LoaderVersionExtractor.ARTIFACT_ID_MAVEN_SUREFIRE_PLUGIN);
+        CustomProviderInfo customProviderInfo = new CustomProviderInfo(temporaryFolder.getRoot());
+
+        // when
+        customProviderInfo.retrieveCustomProviderInformation(temporaryFolder.getRoot(), null);
+
+        // then
+        verifyCustomProviderInfoHasJunit5(customProviderInfo);
+    }
+
+    private void verifyCustomProviderInfoHasJunit5(CustomProviderInfo customProviderInfo) {
+        assertThat(customProviderInfo.getDepCoordinates()).isEqualTo(junit5Coordinates);
+        assertThat(customProviderInfo.getProviderClassName()).isEqualTo(JUNIT_5.getProviderClassName());
```

Comment:
I would inline it or even better - create a custom assertion. Add to the cleanup task I suggested earlier if you want.